src/napari/_vispy/canvas.py

```diff
@@ -175,11 +176,15 @@ def __init__(
         self.viewer.events.theme.connect(
             self._on_theme_change, position='first'
         )
+        # Connecting events from Camera
         self.viewer.camera.events.mouse_pan.connect(self._on_interactive)
         self.viewer.camera.events.mouse_zoom.connect(self._on_interactive)
         self.viewer.camera.events.zoom.connect(self._on_cursor)
+        # Connecting events from Layers
         self.viewer.layers.events.reordered.connect(self._reorder_layers)
         self.viewer.layers.events.removed.connect(self._remove_layer)
+        # Connecting events from Overlays
+        self.viewer._zoom_box.events.zoom.connect(self._on_boxzoom)
         self.viewer._overlays.events.added.connect(
```

Comment:
Sorry to be nitpicky, but I find this kind of comments just noisy when reading :sweat_smile: the code is perfectly clear by itself. Blank lines in between would do a better job.